lib/spack/spack/installer.py

```diff
@@ -1716,6 +1716,16 @@ def install(self):
                 spack.hooks.on_install_cancel(task.request.pkg.spec)
                 raise
 
+            except binary_distribution.NoChecksumException as exc:
+                if not task.request.install_args.get('cache_only'):
+                    # Checking hash on downloaded binary failed.
+                    err = 'Failed to install {0} due to {1}:'
+                    err += ' Requeue for manual installation.'
+                    tty.error(err.format(pkg.name, exc.__class__.__name__))
```

Comment:
Is it the exception name you want here or the error message?

```suggestion
                    tty.error(err.format(pkg.name, str(exc)))
```

Reply:
Error message is probably better. Thanks!